src/main/java/gui/EditRulePopup.java

```diff
@@ -0,0 +1,283 @@
+
+		clearButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				clearMetricsListPanel();
+			}
+		});
+
+		deleteButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					// Add functionality here.
+				}
+			}
+		});
+
+		saveButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					// Add functionality here.
+				}
+			}
+		});
+
+		controlPanel.add(clearButton);
+		controlPanel.add(deleteButton);
+		controlPanel.add(saveButton);
+		return controlPanel;
+
+	}
+
+	/**
+	 * This method both returns and creates the JPanel holding the line which allows
+	 * users to add new metrics into the metrics list. TODO: The currently
+	 * hard-coded values of the JBoxes are to be replaced with the correct, final
+	 * values once the ENUMs are created.
+	 * 
+	 * @return
+	 */
+	public JPanel createAddMetricPanel() {
+		addNewMetricPanel.removeAll();
+		addNewMetricPanel.setLayout(new GridLayout(1, 6));
+		condition = new JComboBox<>();
+		setConditionVisibility();
+		JLabel ifCondition = new JLabel("IF", SwingConstants.CENTER);
+		JComboBox<String> value = new JComboBox<>();
+		value.addItem("LOC");
+		value.addItem("MET");
+
+		JComboBox<String> comparison = new JComboBox<>();
+		comparison.addItem(">");
+		comparison.addItem("<");
+		comparison.addItem("=");
+		comparison.addItem("!=");
+
+		JTextField threshold = new JTextField("10");
+
+		JButton addMetricButton = new JButton("Add");
+		addMetricButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (value.getItemCount() == 0) {
+					JOptionPane.showMessageDialog(null, "Please check if all metric fields were entered correctly!");
+				} else {
+					String metric;
+					if (ruleMetrics.isEmpty()) {
+						metric = "IF " + value.getSelectedItem() + " " + comparison.getSelectedItem() + " "
+								+ threshold.getText();
+
+					} else {
+						metric = condition.getSelectedItem() + " " + value.getSelectedItem() + " "
+								+ comparison.getSelectedItem() + " " + threshold.getText();
+					}
+					ruleMetrics.add(metric);
+					fillMetricsListPanel();
+					setConditionVisibility();
+					metricsListPanel.revalidate();
+					metricsListPanel.repaint();
+				}
+			}
+		});
+
+		addNewMetricPanel.add(condition);
+		addNewMetricPanel.add(ifCondition);
+		addNewMetricPanel.add(value);
+		addNewMetricPanel.add(comparison);
+		addNewMetricPanel.add(threshold);
+		addNewMetricPanel.add(addMetricButton);
+		addNewMetricPanel.setPreferredSize(new Dimension(650, 25));
+		addNewMetricPanel.setMaximumSize(new Dimension(650, 25));
+		return addNewMetricPanel;
+	}
+
+//TODO

```

Comment:
What is there to do here? you didn't write anything

Reply:
The TODO's accompanying test was written in the javadocs right bellow it. I will put the TODO in said javadocs por clarity.